core/lib/web3_decl/src/error.rs

```diff
@@ -42,3 +43,65 @@ pub enum Web3Error {
     #[error("Tree API is not available")]
     TreeApiUnavailable,
 }
+
+use std::{collections::HashMap, error, fmt, fmt::Debug};
+
+#[derive(Debug)]
+pub struct RpcErrorWithDetails {
+    pub inner_error: RpcError,
+    pub method: String,
+    pub args: HashMap<String, String>,
+}
+impl error::Error for RpcErrorWithDetails {}
+
+impl fmt::Display for RpcErrorWithDetails {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let args_formatted: Vec<String> =
+            self.args.iter().map(|(k, v)| format!("{k}: {v}")).collect();
```

Comment:
Why not format `args` with the `Debug` formatter (i.e., `{args:?}`)? It outputs args in a quite similar format.

Reply:
I wanted to make it a bit nicer, but it does make sense to just use Debug formatter